dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,569 @@
+                    .setProperty("originalStartDate", new Date())
+                    .setProperty("recurrenceStart", new Date())
+                    .setProperty("recurrenceEnd", new Date()).nextPersisted();
+
+            validateMappingForFields(testCase, null, fields, expectedResult);
+
+        }finally {
+            if (event != null){
+                ContentletDataGen.destroy(event);
+            }
+        }
+    }
+
+    /**
+     * Validates asserts for tests: testValidateNewsLikeMapping, testValidateFileAssetMapping and testValidateEventMapping
+     * @param testCase
+     * @param contentTypeVarName
+     * @param fields
+     * @param expectedResult
+     * @throws IOException
+     * @throws DotDataException
+     */
+    private void validateMappingForFields(final String testCase, final String contentTypeVarName,  final String[] fields, final String expectedResult)
+            throws IOException, DotDataException {
+        Map<String, String> mapping;
+
+        for (final String field : fields) {
+            Logger.info(this,
+                    "Validating mapping for case: " + testCase + ". Field Name: " + field);
+            mapping = (Map<String, String>) esMappingAPI
+                    .getFieldMappingAsMap(APILocator.getIndiciesAPI().loadIndicies().getWorking(),
+                            expectedResult.equals("geo_point") ? contentTypeVarName
+                                    + StringPool.PERIOD + field : field).entrySet().iterator()
+                    .next().getValue();
+            assertTrue(UtilMethods.isSet(mapping.get("type")));
+            assertEquals(expectedResult, mapping.get("type"));
+            Logger.info(this,
+                    "Case: " + testCase + ". Field Name: " + field + " validated successfully");
+        }
+    }
+
+
+    /**
+     * Creates a new Legacy Relationship object
+     * @param parentContentType
+     * @param childContentType
+     * @return
+     */
+    private Relationship createLegacyRelationship(final ContentType parentContentType,
+            final ContentType childContentType) {
+        final String relationTypeValue = parentContentType.name() + "-" + childContentType.name();
+
+        Relationship relationship;
+        relationship = relationshipAPI.byTypeValue(relationTypeValue);
+        if (null != relationship) {
+            return relationship;
+        } else {
+            relationship = new Relationship();
+            relationship.setParentRelationName(parentContentType.name());
+            relationship.setChildRelationName(childContentType.name());
+            relationship.setCardinality(0);
+            relationship.setRelationTypeValue(relationTypeValue);
+            relationship.setParentStructureInode(parentContentType.inode());
+            relationship.setChildStructureInode(childContentType.id());
+            try {
+                relationshipAPI.create(relationship);
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)